server/internal/core/application/service.go

```diff
@@ -835,29 +850,29 @@ func (s *covenantlessService) RegisterIntent(ctx context.Context, bip322signatur
 					return "", fmt.Errorf("tx %s not confirmed", vtxoKey.Txid)
 				}
 
-				vtxoScript, err := tree.ParseVtxoScript(tapscripts)
-				if err != nil {
-					return "", fmt.Errorf("failed to parse boarding utxo taproot tree: %s", err)
-				}
-
 				// validate the vtxo script
 				if err := vtxoScript.Validate(s.pubkey, common.RelativeLocktime{
 					Type:  s.boardingExitDelay.Type,
 					Value: s.boardingExitDelay.Value,
-				}); err != nil {
+				}, s.allowCSVBlockType); err != nil {
 					return "", fmt.Errorf("invalid vtxo script: %s", err)
 				}
 
-				exitDelay, err := vtxoScript.SmallestExitDelay()
-				if err != nil {
-					return "", fmt.Errorf("failed to get exit delay: %s", err)
-				}
-
 				// if the exit path is available, forbid registering the boarding utxo
 				if time.Unix(blocktime, 0).Add(time.Duration(exitDelay.Seconds()) * time.Second).Before(time.Now()) {
 					return "", fmt.Errorf("tx %s expired", vtxoKey.Txid)
 				}
 
+				// If the intent is registered using a exit path that contains CSV delay, we want to verify it
+				// by shifitng the current "now" in the future of the duration of the smallest exit delay.
+				// This way, any exit order guaranteed by the exit path is maintained at intent registration
+				if !disabled {
+					delta := time.Now().Add(time.Duration(exitDelay.Seconds())*time.Second).Unix() - blocktime
```

Comment:
can we take a snapshot of time.Now in the beginning so that no weird conditions happen midway through

Reply:
done